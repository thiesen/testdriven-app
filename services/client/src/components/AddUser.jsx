import React from 'react';

const AddUser = props => (
    <form>
      <div className="form-group">
        <input
          name="username"
          className="form-control input-lg"
          type="text"
          placeholder="Enter an username"
          required
          />
        <input
          name="email"
          className="form-control input-lg"
          type="email"
          placeholder="Enter an email address"
          required
          />
      </div>
      <input
        type="submit"
        className="btn btn-primary btn-lg btn-block"
        value="Submit"
        />
    </form>
);

export default AddUser;
