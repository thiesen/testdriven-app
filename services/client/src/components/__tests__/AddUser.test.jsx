import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import AddUser from '../AddUser';


describe('<AddUser />', () => {
  it('renders a form with expected fields', () => {
    const wrapper = shallow(<AddUser />);
    const element = wrapper.find('form');

    expect(element.find('input').length).toBe(3);
    expect(element.find('input').get(0).props.name).toEqual('username');
    expect(element.find('input').get(1).props.name).toEqual('email');
    expect(element.find('input').get(2).props.type).toEqual('submit');

  });

  test('snapshot', () => {
    const tree = renderer.create(<AddUser/>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
