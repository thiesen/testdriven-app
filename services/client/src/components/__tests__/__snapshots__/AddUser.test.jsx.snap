// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddUser /> snapshot 1`] = `
<form>
  <div
    className="form-group"
  >
    <input
      className="form-control input-lg"
      name="username"
      placeholder="Enter an username"
      required={true}
      type="text"
    />
    <input
      className="form-control input-lg"
      name="email"
      placeholder="Enter an email address"
      required={true}
      type="email"
    />
  </div>
  <input
    className="btn btn-primary btn-lg btn-block"
    type="submit"
    value="Submit"
  />
</form>
`;
